### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by setrofim**
Add demo frontend.

### CODE REVIEW

File: plugins/opapolicyengine/main_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	endorsementsPath := filepath.Join(wd, "test", "endorsements.json")
+	evidencePath := filepath.Join(wd, "test", "evidence.json")
+	policyPath := filepath.Join(wd, "test", "policy.rego")
+
+	endorsements, err := readJSONObjectFromFile(endorsementsPath)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:42]:
```suggestion
	require.Nil(err)
```

File: plugins/opapolicyengine/main_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	evidence, err := readJSONObjectFromFile(evidencePath)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:47]:
```suggestion
	require.Nil(err)
```

File: plugins/opapolicyengine/main_test.go [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+	policyData, err := ioutil.ReadFile(policyPath)
+	if err != nil {
+		t.Fatalf("%v", err)
+	}
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:52]:
```suggestion
	require.Nil(err)
```

File: plugins/opapolicyengine/main_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+	var pe OpaPolicyEngine
+	var peParams common.PolicyEngineParams
+
+	if err = pe.Init(peParams); err != nil {
+		t.Fatalf("%v", err)
+	}
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:59]:
```suggestion
	err = pe.Init(peParams)
	require.Nil(err)
```

File: plugins/opapolicyengine/main_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+	if err = pe.LoadPolicy(policyData); err != nil {
+		t.Fatalf("%v", err)
+	}
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:63]:
```suggestion
	err = pe.LoadPolicy(policyData)
	require.Nil(err)
```

File: plugins/opapolicyengine/main_test.go [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+	isValid, err := pe.CheckValid(evidence, endorsements)
+	require.Nil(err)
+	require.True(isValid)
```
Comment (thomas-fossati) [inline plugins/opapolicyengine/main_test.go:67]: These two look more like `assert`s to me

(The idea is to use `require` for test preconditions and `assert` for test assertions.)

File: plugins/sqlitepolicy/main_test.go [hunk 7]
```diff
@@ -56,33 +56,33 @@ func finiDb(path string) {
...
 
 func TestSqliteGetPolicy(t *testing.T) {
-	assert := assert.New(t)
+	require := require.New(t)
 
 	wd, err := os.Getwd()
 	if err != nil {
-		t.Errorf("%v", err)
+		require.FailNow("%v", err)
 	}
```
Comment (thomas-fossati) [inline plugins/sqlitepolicy/main_test.go:64]:
```suggestion
	require.Nil(err)
```

File: plugins/sqlitepolicy/main_test.go [hunk 8]
```diff
@@ -56,33 +56,33 @@ func finiDb(path string) {
 }
 
 func TestSqliteGetPolicy(t *testing.T) {
-	assert := assert.New(t)
+	require := require.New(t)
 
 	wd, err := os.Getwd()
 	if err != nil {
```
Comment (thomas-fossati) [inline plugins/sqlitepolicy/main_test.go:62]:
```suggestion
```

File: plugins/sqlitepolicy/main_test.go [hunk 9]
```diff
@@ -56,33 +56,33 @@ func finiDb(path string) {
...
-		t.Errorf("%v", err)
+		require.FailNow("%v", err)
 	}
 
 	schemaFile := filepath.Join(wd, "test", "iat-policy.sqlite")
 	dbPath, err := initDb(schemaFile)
 	if err != nil {
```
Comment (thomas-fossati) [inline plugins/sqlitepolicy/main_test.go:68]: same as above

File: plugins/sqlitepolicy/main_test.go [hunk 10]
```diff
@@ -56,33 +56,33 @@ func finiDb(path string) {
...
-		t.Errorf("%v", err)
+		require.FailNow("%v", err)
 	}
 	defer finiDb(dbPath)
 
 	var pm PolicyStore
 	err = pm.Init(common.PolicyStoreParams{"dbPath": dbPath})
 	if err != nil {
```
Comment (thomas-fossati) [inline plugins/sqlitepolicy/main_test.go:75]: ditto

File: frontend/Makefile [hunk 11]
```diff
@@ -0,0 +1,17 @@
+include ../mk/common.mk
+include ../mk/lint.mk
+include ../mk/test.mk
+
+.PHONY: all
+all: verifier
+
+verifier:
+	go build -o verifier cmd/veraison/main.go
```
Comment (thomas-fossati) [inline frontend/Makefile:9]: without an explicit dependency on the source, the rule becomes quickly inert
```suggestion
verifier: cmd/veraison/main.go
	go build -o $@ $<
```

**Review by thomas-fossati — CHANGES_REQUESTED** [review]
my computer is going crazy

File: frontend/cmd/veraison/main.go [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+func main() {
+	var pluginDir, dbPath string
+
+	flag.StringVar(&pluginDir, "p", "", "Path to directory containing plugin binaries.")
+	flag.StringVar(&dbPath, "d", "", "Path to the directory containing sqlite database files.")
+	flag.Parse()
+
+	logger, err := zap.NewDevelopment() // TODO configurable
```
[the remark this answers is not in the document]
Comment (setrofim) [reply frontend/cmd/veraison/main.go:18]: Done: https://github.com/veraison/veraison/issues/32

**Comment by setrofim** [comment]
Review feedback addressed.

**Review by thomas-fossati — APPROVED** [review]
LGTM, thanks for addressing my tedious comments!

**Comment by setrofim** [comment]
> just a curiosity: do we still need the lint annotation now that we use reportProblem?

You're totally right, they're no longer needed. I've removed them.

**Pull request merged**